src/NServiceBus.Core/Unicast/Transport/Config/TransportExtensions.cs

```diff
@@ -28,15 +28,36 @@ public new TransportExtensions<T> ConnectionString(string connectionString)
             return this;
         }
 
+#if NET452
         /// <summary>
         /// Configures the transport to use the connection string with the given name.
         /// </summary>
+        [ObsoleteEx(
+        Message = "Using named connection strings is discouraged. Instead, load the connection string in your code and pass the value to EndpointConfiguration.UseTransport().ConnectionString(connectionString).",
+            ReplacementTypeOrMember = "TransportExtensions<T>.ConnectionString(connectionString)",
```

Comment:
Just a note that `ReplacementTypeOrMember` should only be used when there is a straight replacement. Not blocking the PR